wasm/src/programs/macros.rs

```diff
@@ -136,33 +136,55 @@ macro_rules! execute_fee {
                 $process.authorize_fee_private::<CurrentAleo, _>(
                     $private_key,
                     fee_record_native,
+                    $minimum_execution_cost,
                     $fee_microcredits,
-                    0u64,
                     $execution_id,
                     $rng,
                 ).map_err(|e| e.to_string())?
             }
             None => {
-                $process.authorize_fee_public::<CurrentAleo, _>($private_key, $fee_microcredits, 0u64, $execution_id, $rng).map_err(|e| e.to_string())?
+                $process.authorize_fee_public::<CurrentAleo, _>(
+                    $private_key,
+                    $minimum_execution_cost,
+                    $fee_microcredits,
+                    $execution_id,
+                    $rng,
+                ).map_err(|e| e.to_string())?
             }
         };
 
-        log("Executing fee");
+        log("Executing Fee");
         let (_, mut trace) = $process
             .execute::<CurrentAleo, _>(fee_authorization, $rng)
-            .map_err(|err| err.to_string())?;
+            .map_err(|e| e.to_string())?;
 
-        if let Some(offline_query) = $offline_query {
-            trace.prepare_async(offline_query).await.map_err(|err| err.to_string())?;
-        } else {
-            let query = QueryNative::from($submission_url);
-            trace.prepare_async(query).await.map_err(|err| err.to_string())?;
-        };
         let fee = trace.prove_fee::<CurrentAleo, _>(&mut StdRng::from_entropy()).map_err(|e|e.to_string())?;
 
         log("Verifying fee execution");
         $process.verify_fee(&fee, $execution_id).map_err(|e| e.to_string())?;
 
         fee
+    }}
+}
+
+#[macro_export]
+macro_rules! calculate_minimum_fee {
```

Comment:
This duplicates the query formation, perhaps this should be renamed to `prepare_trace_and_calculate_fee` and do the trace preparation in here as well.